packages/xstate-solid/src/deriveServiceState.ts

```diff
@@ -0,0 +1,64 @@
+import type {
+  ActorRef,
+  Interpreter,
+  InterpreterFrom,
+  StateLike,
+  State
+} from 'xstate';
+
+export function isStateLike(state: any): state is StateLike<any> {
+  return (
+    typeof state === 'object' &&
+    'value' in state &&
+    'context' in state &&
+    'event' in state &&
+    '_event' in state
+  );
+}
+
+function isService(value: any): value is Interpreter<any> {
+  return 'state' in value && 'machine' in value;
+}
+
+/**
+ * Takes in an interpreter or actor ref and returns a State object with reactive
+ * methods or if not State, the initial value passed in
+ * @param service {InterpreterFrom<any> | ActorRef<any>}
+ * @param initialState {State<any> | unknown}
+ */
+export const deriveServiceState = <
+  TService extends InterpreterFrom<any> | ActorRef<any>,
+  InitialState extends State<any>
+>(
+  service: TService,
+  initialState: InitialState extends unknown ? unknown : InitialState
```

Comment:
I would rename this to `state` or something - it's used for way more than just the initial state.

Reply:
I also find this conditional type confusing - we gonna always end up with `unknown` here as everything `extends unknown`